GoogleForCreators/web-stories-wp
# Editor: Support publishing pending stories

## Context



It wasn't previously possible for admins to publish a pending story through the editor.

## Summary



This change gives admin users the expected ability to publish pending stories, or revert them to regular drafts again.

## Relevant Technical Choices



- Some slight refactorings to make code better readable

## To-do



## User-facing changes



Contributor users will have two buttons: Save (as draft) and Submit for review:

![Screenshot 2021-11-26 at 15 38 37](https://user-images.githubusercontent.com/841956/143596844-37be3c31-47f6-42cf-b32a-f3d9c0311423.png)

Admins, when editing a pending story will have three buttons: Switch to draft, save (as pending), publish

![Screenshot 2021-11-26 at 15 38 32](https://user-images.githubusercontent.com/841956/143596940-362e6e7d-40b9-401f-9908-8049fd8d15b1.png)

## Testing Instructions






This PR can be tested by following these steps:

1. As a contributor user, create a new story
2. Submit it for review
3. Log in as admin user and edit the pending story
4. Verify that the story can be saved, switched to draft, or published

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #9798
